src/react/components/common/tagInput/tagInput.tsx

```diff
@@ -463,19 +468,34 @@ export class TagInput extends React.Component<ITagInputProps, ITagInputState> {
             let deselect = selected && oldTagOperation === TagOperationMode.None;
 
             // Only fire click event if a region is selected
-            if (this.props.selectedRegions &&
-                this.props.selectedRegions.length > 0 &&
-                this.props.onTagClick) {
-                deselect = false;
-                this.props.onTagClick(tag);
+            const { selectedRegions, onTagClick } = this.props;
+            if (selectedRegions && selectedRegions.length && onTagClick) {
+                const { category } = selectedRegions[0];
+                const { format, type, documentCount } = tag;
+                const tagCategory = this.getTagCategory(type);
+                if (tagCategory === category ||
+                    (documentCount === 0 && type === FieldType.String && format === FieldFormat.NotSpecified)) {
+                        onTagClick(tag);
+                        deselect = false;
+                } else {
+                    toast.warn(strings.tags.warnings.notCompatibleTagType);
+                }
             }
-
             this.setState({
                 selectedTag: deselect ? null : tag,
                 tagOperation,
             });
 
-       }
+        }
+    }
+
+    private getTagCategory = (tagType: string) => {
+        switch (tagType) {
+            case FieldType.SelectionMark:
```

Comment:
This way, in the future when we add another non-compatible type (like checkbox), we can just add a case to the switch statement